src/Chapter2.hs

```diff
@@ -727,7 +745,9 @@ value of the element itself
 🕯 HINT: Use combination of 'map' and 'replicate'
 -}
 smartReplicate :: [Int] -> [Int]
-smartReplicate l = error "smartReplicate: Not implemented!"
+smartReplicate [] = []
+smartReplicate (x:xs) = replicate x x ++ smartReplicate xs
```

Comment:
Nice one!

Note, that you also can use standard functions to do the same. You see that you actually map `replicate` over each element of the list and then concatenate all the results together. You can take a look at `concatMap` standard function, which behaves exactly like this! 🙂